ci/deploy/npm.py

```diff
@@ -0,0 +1,164 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import argparse
+import json
+import logging
+import os
+import shutil
+import urllib.parse
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional
+
+import requests
+
+from materialize import spawn
+
+logging.basicConfig(
+    format="%(asctime)s [%(levelname)s] %(message)s",
+    level=os.environ.get("MZ_DEV_LOG", "INFO").upper(),
+)
+logger = logging.getLogger(__name__)
+
+PROJECT_PATHS = ["src/sql-lexer"]
+ROOT_PATH = Path(".")
+
+
+@dataclass(frozen=True)
+class Version:
+    rust: str
+    node: str
+    is_development: bool
+
+
+def generate_version(rust_version: str, build_identifier: Optional[int]) -> Version:
+    node_version = rust_version
+    is_development = False
+    if rust_version.endswith("-dev"):
+        if build_identifier is None:
+            raise ValueError(
+                "a build identifier must be provided for prerelease builds"
+            )
+        node_version = f"{crate_version}.{build_identifier}"
+        is_development = True
+    return Version(rust=rust_version, node=node_version, is_development=is_development)
+
+
+def get_crate_version() -> str:
+    with open("src/environmentd/Cargo.toml", "r") as envd:
+        for line in envd:
+            if line.startswith("version ="):
+                return line.split("=").pop().strip().replace('"', "")
+    raise ValueError("version not found")
+
+
+def build_package(version: Version, project_path: Path) -> Path:
+    spawn.runv(["bin/wasm-build", str(project_path)])
+    package_path = project_path / "pkg"
```

Comment:
nit: I think slightly less confusing to use `crate_path` instead of `proect_path`.

```suggestion
def build_package(version: Version, crate_path: Path) -> Path:
    spawn.runv(["bin/wasm-build", str(crate_path)])
    package_path = crate_path / "pkg"
```